.github/workflows/cargo-deny-pr.yml

```diff
@@ -0,0 +1,24 @@
+name: cargo deny
+on:
+  push:
+    paths:
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+  pull_request:
+    paths:
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+jobs:
+  cargo-deny:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        checks:
+          - advisories
+          - bans licenses sources
```

Comment:
Given that we no longer treat `advisories` special, is the matrix still needed? Also do we need to list the commands any longer at all? Wouldn't a `cargo deny check` suffice to run them all?

Reply:
I personally prefer to be explicit as if there is change in the future you can gradually adopt more check features.

If the preference is to be implicit non-exhaustive then we can change this but prepare for surprise(s) if the tooling changes.

Let me know and I can change.

Reply:
I'd like to keep it separate actually so we can temporarily ignore a failing advisory check for example. It also allows us to mark some of them as required via GH for merging and leave others optional.